src/d3a/models/market/market_structures.py

```diff
@@ -161,10 +181,12 @@ def to_JSON_string(self):
 
 def trade_from_JSON_string(trade_string):
     trade_dict = json.loads(trade_string)
-    trade_dict['offer'] = offer_from_JSON_string(trade_dict['offer'])
+    trade_dict['offer'] = offer_or_bid_from_JSON_string(trade_dict['offer'])
     if 'residual' in trade_dict and trade_dict['residual'] is not None:
-        trade_dict['residual'] = offer_from_JSON_string(trade_dict['residual'])
+        trade_dict['residual'] = offer_or_bid_from_JSON_string(trade_dict['residual'])
     trade_dict['time'] = pendulum.parse(trade_dict['time'])
+    # if 'offer_bid_trade_info' in trade_dict:
```

Comment:
Any reason for this to be removed?

Reply:
Yes, the reason for that is because this is a list with not always 5 arguments, but also 3. If you want me to give it a deeper look on that let me know and I will find out the reason.

Reply:
Sorry, I am confused. Which list are you referring to? But i just realise, the we dont serialise the `offer_bid_trade_info` member of the trade. It should be added in the to_JSON_string method, right?

Reply:
Not really. The type of `trade_dict['offer_bid_trade_info']` is not a TradeBidInfo, but a list with 3 elements (have a look at grid_fees/base_model.py, method propagate_original_bid_info_on_offer_trade or propagate_original_offer_info_on_bid_trade). The reason it was implemented that way was that either the bid or the offer members need to be populated according to the trade (if it is an accept_offer, the offer members will be populated, if it is an accept_bid the bid members will be populated). Even though this works, I will refactor it to use the same type and to do sanity checks whether the correct members are None.

Reply:
Ah, yes! Sorry. If this works for now, i susggest: dont do it in this story but open another story please.

Reply:
Enhancement story created https://gridsingularity.atlassian.net/browse/D3ASIM-1742